floodviz/map_utils.py

```diff
@@ -52,26 +55,61 @@ def site_dict(site_list, url_prefix):
     return dnice
 
 
-def get_sites_geojson(data):
-    """Writes site data to a .json file so it can be mapped
-       Args:
-           data: A list of dicts with data to be written to the file
-        Returns:
-            A string containing geojson data
+def write_geojson(filename, data):
     """
+    Writes site data to a .json file so it can be mapped
+    This version will create the file if it does not exist and overwrite it if it does. You have been warned.
 
-    geojson = "{ \"type\": \"FeatureCollection\", \"features\": [ \n"
-
-    for i, datum in enumerate(data):
-        geojson += "{ \"type\": \"Feature\",\n \"geometry\": {\n \"type\": \"Point\",\n \"coordinates\" : " \
-                       "[" + datum.get('dec_long_va', 'n/a') + ", " + datum.get('dec_lat_va', 'n/a') \
-                   + "]\n },\n \"properties\": {\n \"name\": \"" + datum.get('station_nm', 'n/a') \
-                   + "\",\n \"id\": \"" + datum.get('site_no', 'n/a') + "\",\n \"huc\": \"" \
-                   + datum.get('huc_cd', 'n/a') + "\" \n } \n }"
-        # add a comma unless at end of list
-        if data[i] != data[len(data) - 1]:
-            geojson += ","
-        geojson += "\n"
-    geojson += " ] }"
-
-    return geojson
\ No newline at end of file
+    :param  filename: the file to be written to
+    :param  data: the data to be written to the file
+    """
+    data = create_geojson(data)
+    with open(filename, "w+") as file:
+        json.dump(data, file, indent=True)
+
+
+def create_geojson(data):
+    """
+    :param data: a dictionary of sites to be transformed into geojson
+    :return: a geojson string representing the sites
+    """
+
+    new_data = {
+        "type": "FeatureCollection",
+        "features": []
+    }
+    for original in data:
+        item = {
+            'type': 'Feature',
+            'geometry': {
+                'type': 'Point',
+                'coordinates': [
+                    original['dec_long_va'],
+                    original['dec_lat_va']
+                ]
+            },
+            'properties': {
+                'name': original['station_nm'],
+                'id': original['site_no'],
+                'huc': original['huc_cd']
+            }
+        }
+        new_data['features'].append(item)
+    return new_data
+
+
+def projection_info(code, url):
```

Comment:
we should add tests for this. Don't won't to hold up merging for that. Add the tests when you have time.